### PR TITLE
Add batch_training, resume training from checkpoint, and more optimizer/scheduler options

### PR DESCRIPTION
**Pull request by aaprasad**
* Added batch_training as an option to `train.py` if user calls it as `train.py +base_params=base.yaml +batch_cfg=hparams.csv`
* Added ability to resume from checkpoint
* moved initialization of GTR model + loss into LightningModule to reduce checkpoint file size
* added more flexibility in optimizer/scheduler choice by using an init function to retrieve optimizer and scheduler based on config

### CODE REVIEW

File: biogtr/models/global_tracking_transformer.py [hunk 1]
```diff
@@ -28,6 +28,7 @@ def __init__(
         embedding_meta: dict = None,
         return_embedding: bool = False,
         decoder_self_attn: bool = False,
+        **kwargs,
```
Comment (sheridana) [inline biogtr/models/global_tracking_transformer.py:31]: Should be fine, but let's come back to this - maybe we don't want to use kwargs here.

File: biogtr/training/train.py [hunk 2]
```diff
@@ -27,26 +34,45 @@ def main(cfg: DictConfig):
...
         cfg: The config dict parsed by `hydra`
     """
     train_cfg = Config(cfg)
-
+    pprint(f"Base train config: {train_cfg.cfg}")
+
+    # update with parameters for batch train job
+    if "batch_config" in cfg.keys():
+        try:
+            index = int(os.environ["POD_INDEX"])
```
Comment (sheridana) [inline biogtr/training/train.py:48]: This assumes the hparams csv is already created - we should also add ability to create the csv given desired grid search hparams.

**Review by sheridana — COMMENTED** [review]
Minor comments

File: biogtr/training/train.py (hunk 2 above)
Comment (aaprasad) [reply biogtr/training/train.py:48]: TODO: Come back to this

File: biogtr/models/global_tracking_transformer.py (hunk 1 above)
Comment (aaprasad) [reply biogtr/models/global_tracking_transformer.py:31]: keep as is for now may change later

**Pull request merged**